src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilder.cs

```diff
@@ -64,53 +64,14 @@ public ClusterBuilder()
             _eventAggregator = new EventAggregator();
         }
 
-        // methods
+        // public methods
         /// <summary>
         /// Builds the cluster.
         /// </summary>
         /// <returns>A cluster.</returns>
         public ICluster BuildCluster()
         {
-            IStreamFactory streamFactory = new TcpStreamFactory(_tcpStreamSettings);
```

Comment:
This method had gotten way too big to understand easily and was going to get even bigger with the changes needed, so it has been refactored into a set of smaller helper methods.

Reply:
Wow, nice refactor 😸